src/content-steering-controller.js

```diff
@@ -234,44 +273,91 @@ export default class ContentSteeringController extends videojs.EventTarget {
     this.steeringManifest.priority = steeringJson['PATHWAY-PRIORITY'] || steeringJson['SERVICE-LOCATION-PRIORITY'];
     // TODO: HLS handle PATHWAY-CLONES. See section 7.2 https://datatracker.ietf.org/doc/draft-pantos-hls-rfc8216bis/
 
-    // TODO: fully implement priority logic.
     // 1. apply first pathway from the array.
-    // 2. if first first pathway doesn't exist in manifest, try next pathway.
+    // 2. if first pathway doesn't exist in manifest, try next pathway.
     //    a. if all pathways are exhausted, ignore the steering manifest priority.
     // 3. if segments fail from an established pathway, try all variants/renditions, then exclude the failed pathway.
     //    a. exclude a pathway for a minimum of the last TTL duration. Meaning, from the next steering response,
     //       the excluded pathway will be ignored.
-    const chooseNextPathway = (pathways) => {
-      for (const path of pathways) {
+    //       See excludePathway usage in excludePlaylist().
+
+    // If there are no available pathways, we need to stop content steering.
+    if (!this.availablePathways_.size) {
+      this.logger_('There are no available pathways for content steering. Ending content steering.');
+      this.trigger('error');
+      this.dispose();
+    }
+
+    const chooseNextPathway = (pathwaysByPriority) => {
+      for (const path of pathwaysByPriority) {
         if (this.availablePathways_.has(path)) {
           return path;
         }
       }
+
+      // If no pathway matches, ignore the manifest and choose the first available.
+      return [...this.availablePathways_][0];
```

Comment:
Should we handle here the case when `this.availablePathways_` is empty?

Reply:
I believe we do this a few lines above with the check `if (!this.availablePathways_.size)`